File: examples/go-example/greet_test.go
package greet

import "testing"

func TestGreet(t *testing.T) {
	cases := []struct {
		name  string
		input string
		want  string
	}{
		{name: "named", input: "GoldenCI", want: "Hello, GoldenCI!"},
		{name: "empty falls back to world", input: "", want: "Hello, world!"},
	}

	for _, tc := range cases {
		t.Run(tc.name, func(t *testing.T) {
			if got := Greet(tc.input); got != tc.want {
				t.Errorf("Greet(%q) = %q, want %q", tc.input, got, tc.want)
			}
		})
	}
}

File: examples/node-example/greet.test.js
const { greet } = require('./greet');

test('greets a named caller', () => {
  expect(greet('GoldenCI')).toBe('Hello, GoldenCI!');
});

test('falls back to the world when the name is empty', () => {
  expect(greet('')).toBe('Hello, world!');
});
